docs/developers/contributing.md

```diff
@@ -9,69 +9,109 @@ If you are a new user, we recommend checking out the detailed [Github Docs](http
+    python -m venv <path-to-env>
+    source <path-to-env>/bin/activate
+    ```
+    See the [venv](https://docs.python.org/3/library/venv.html) documentation for instructions on Windows.
+    :::
 
-```sh
-pre-commit install
-```
+    ::::
+
+    ```{note}
+    It is highly recommended to create a fresh environment when working with
+    napari, to prevent issues with outdated or conflicting packages in your
+    development environment.
+    ```
+
+4. Install the package in editable mode, along with all of the developer tools.
+
+    If you only want to use napari, you can install it on most macOS, Linux and
+    Windows systems with Python {{ python_version_range }}
+    by following the directions on the
+    [instructions page](../tutorials/fundamentals/installation.md#install-as-python-package-recommended).
+
+    napari supports different Qt backends, and you can choose which one to install and use.
+
+    For example, for PyQt5, the default, you would use the following:
+    ```sh
+    pip install -e ".[pyqt,dev]"  # (quotes only needed for zsh shell)
+    ```
 
-Upon committing, your code will be formatted according to our [`black`
-configuration](https://github.com/napari/napari/blob/main/pyproject.toml), which includes the settings
-`skip-string-normalization = true` and `max-line-length = 79`. To learn more,
-see [`black`'s documentation](https://black.readthedocs.io/en/stable/).
-
-Code will also be linted to enforce the stylistic and logistical rules specified
-in our [`flake8` configuration](https://github.com/napari/napari/blob/main/setup.cfg), which currently ignores
-[E203](https://lintlyci.github.io/Flake8Rules/rules/E203.html),
-[E501](https://lintlyci.github.io/Flake8Rules/rules/E501.html),
-[W503](https://lintlyci.github.io/Flake8Rules/rules/W503.html) and
-[C901](https://lintlyci.github.io/Flake8Rules/rules/C901.html).  For information
-on any specific flake8 error code, see the [Flake8
-Rules](https://lintlyci.github.io/Flake8Rules/).  You may also wish to refer to
-the [PEP 8 style guide](https://peps.python.org/pep-0008/).
-
-If you wish to tell the linter to ignore a specific line use the `# noqa`
-comment along with the specific error code (e.g. `import sys  # noqa: E402`) but
-please do not ignore errors lightly.
+    If you want to use PySide2 instead, you would use:
+    ```sh
+    pip install -e ".[pyside,dev]"  # (quotes only needed for zsh shell)
+    ```
+
+    Finally, if you already have a Qt backend installed or want to use an experimental one like Qt6 use:
+    ```sh
+    pip install -e ".[dev]"  # (quotes only needed for zsh shell)
+    ```
+
+    Note that in this last case you will need to install your Qt backend separately.
+
+5. We use [`pre-commit`](https://pre-commit.com) to sort imports with
+   [`isort`](https://github.com/PyCQA/isort), format code with
+   [`black`](https://github.com/psf/black), and lint with
+   [`flake8`](https://github.com/PyCQA/flake8) automatically prior to each commit.
```

Comment:
```suggestion
5. We use [`pre-commit`](https://pre-commit.com) format code with
   [`black`](https://github.com/psf/black), and lint with
   [`ruff`](https://github.com/charliermarsh/ruff) automatically prior to each commit.
```

Reply:
@Czaki 
Should be:
We use pre-commit `to` format code with black`<delete COMMA>` and lint with ...`

Reply:
@psobolewskiPhD fixed